internal/vault/secrets/ephemeral/azure_secrets_static_credentials.go

```diff
@@ -0,0 +1,166 @@
+				Required:            true,
+			},
+			consts.FieldRole: schema.StringAttribute{
+				MarkdownDescription: "Static role name to fetch credentials for.",
+				Required:            true,
+			},
+
+			consts.FieldMetadata: schema.MapAttribute{
+				MarkdownDescription: "Metadata to pass along with the read request.",
+				ElementType:         types.StringType,
+				Optional:            true,
+			},
+			consts.FieldClientID: schema.StringAttribute{
+				MarkdownDescription: "Client ID of the Azure application.",
+				Computed:            true,
+			},
+			consts.FieldClientSecret: schema.StringAttribute{
+				MarkdownDescription: "Client secret of the Azure application.",
+				Computed:            true,
+				Sensitive:           true,
+			},
+			consts.FieldSecretID: schema.StringAttribute{
+				MarkdownDescription: "Secret ID of the Azure application.",
+				Computed:            true,
+			},
+			consts.FieldExpiration: schema.StringAttribute{
+				MarkdownDescription: "Credential expiration time (RFC3339).",
+				Computed:            true,
+			},
+			fieldCombinedRoleMetadata: schema.MapAttribute{
+				MarkdownDescription: "Merged metadata returned by Vault (role + request metadata; role takes precedence).",
+				ElementType:         types.StringType,
+				Computed:            true,
+			},
+		},
+		MarkdownDescription: "Ephemeral: reads Azure static credentials from `/<backend>/static-creds/<role>`. " +
+			"`metadata` is input-only and sent with the request; `combined_role_metadata` is the merged map returned by Vault.",
+	}
+	base.MustAddBaseEphemeralSchema(&resp.Schema)
+}
+
+func (r *AzureStaticCredsEphemeralSecretResource) Metadata(_ context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_azure_static_credentials"
+}
+
+func (r *AzureStaticCredsEphemeralSecretResource) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data AzureStaticCredsEphemeralSecretModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	c, err := client.GetClient(ctx, r.Meta(), data.Namespace.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(errutil.ClientConfigureErr(err))
+		return
+	}
+
+	path := fmt.Sprintf("%s/static-creds/%s", data.Backend.ValueString(), data.Role.ValueString())
+
+	var readData map[string][]string
+	if !data.Metadata.IsNull() && !data.Metadata.IsUnknown() {
+		var inMeta map[string]string
+		if md := data.Metadata.ElementsAs(ctx, &inMeta, false); md.HasError() {
+			resp.Diagnostics.Append(md...)
+			return
+		}
+		if len(inMeta) > 0 {
+			kvPairs := make([]string, 0, len(inMeta))
+			for k, v := range inMeta {
+				kvPairs = append(kvPairs, fmt.Sprintf("%s=%s", k, v))
+			}
+			readData = map[string][]string{"metadata": kvPairs}
+		}
+	}
+
+	var sec *api.Secret
+	if readData != nil {
+		sec, err = c.Logical().ReadWithData(path, readData)
```

Comment:
nit: using ctx methods
```suggestion
		sec, err = c.Logical().ReadWithDataWithContext(ctx, path, readData)
```